src/main/java/com/softvision/books/repositories/entities/AbstractEntity.java

```diff
@@ -0,0 +1,50 @@
+package com.softvision.books.repositories.entities;
+
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.*;
+import java.util.Date;
+
+@EntityListeners(AuditingEntityListener.class)
+@MappedSuperclass
+public class AbstractEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id")
+    protected Long id;
```

Comment:
You can remove setting here and set in the AuthorEntity  and BookEntity instead.

You can set the strategy in AuthorEntity  and BookEntity as IDENTITY, such that id of books can also start at 1, regardless if created multiple authors before adding first book to author 1.

@id
@GeneratedValue(strategy = GenerationType.IDENTITY)